crates/circuit/src/operations.rs

```diff
@@ -2860,7 +2848,19 @@ impl Operation for UnitaryGate {
         }
     }
     fn definition(&self, _params: &[Param]) -> Option<CircuitData> {
-        None
+        Python::with_gil(|py| -> Option<CircuitData> {
+            let py_op = self
+                .create_py_op(py, &ExtraInstructionAttributes::default())
+                .unwrap();
+            match py_op.getattr(py, intern!(py, "definition")) {
+                Ok(definition) => definition
+                    .getattr(py, intern!(py, "_data"))
+                    .ok()?
+                    .extract::<CircuitData>(py)
+                    .ok(),
+                Err(_) => None,
```

Comment:
Do we need a definition for UnitaryGate at all? I purposely didn't define it here because the python space one is to call unitary synthesis. In rust it's a bit of a layer violation to do that since it lives in the accelerate crate. While going through python avoids that issue, I'm wondering if it's really necessary because it's not a definition in the strict sense like it is for a concrete gate.

Reply:
Why not just call the two qubit decomposer on the matrix if you encounter a unitary gate in hls during recursion? Realistically that's what you are doing here, just via a very inefficient path that goes back and forth through python.

Reply:
To make sure I understand this correctly, for 1- or 2- qubit unitaries we can use the decomposers already available in rust, and so we can completely avoid going through python; but for larger unitaries the `_define` method of `UnitaryGate` uses `qs_decomposition`, which is only available in python, and so we would still have to go through the python space.

Reply:
This makes sense. So I should revert the change to handle unitary gates? Alternatively, I could close this PR, and only keep the HLS PR, since it's the only consumer for the definition method.

Reply:
Reverted in [5079541](https://github.com/Qiskit/qiskit/pull/13777/commits/50795419a05907c01982f4b9488c76fe0c0a3734).